projects/novo-elements/src/elements/data-table/cell-headers/data-table-header-cell.component.ts

```diff
@@ -283,6 +304,19 @@ export class NovoDataTableCellHeader<T> implements IDataTableSortFilter, OnInit,
     this.multiSelect = this.config.filterConfig && this.config.filterConfig.type ? this.config.filterConfig.type === 'multi-select' : false;
     if (this.multiSelect) {
       this.multiSelectedOptions = this.filter ? [...this.filter] : [];
+      if (typeof this.config.filterConfig.options[1] === 'string') {
+        (this.config.filterConfig.options as string[]).forEach(
```

Comment:
`options` has a well defined type, but you're casting it here in order to narrow the type. TypeScript should see the typeof check and type narrow for you, making the cast unnecessary. In this case though you have to use bracket notation and that's breaking type checking. To avoid casting, move your typeof check inside of the forEach loop:
```
this.config.filterConfig.options.forEach((s: string | IDataTableColumnFilterOption) => {
        if (typeof s === 'string') {
             //string stuff
        } else {
             // IDataTableColumnFilterOption stuff
        }
      });
```
note that filterConfig & options are optional and therefore need undefined checks before accessing them (we do not yet have the strict TS compiler options enabled that should make accesses like this illegal).

Reply:
That actually wasn't working for me, I get ```error TS2349: Cannot invoke an expression whose type lacks a call signature. Type '((callbackfn: (value: string, index: number, array: string[]) => void, thisArg?: any) => void) | ((callbackfn: (value: IDataTableColumnFilterOption, index: number, array: IDataTableColumnFilterOption[]) => void, thisArg?: any) => void)' has no compatible call signatures.```